plotting-visualization/rtdose_visual.py

```diff
@@ -0,0 +1,250 @@
+    ax.imshow(volume[ax.index,:,:],cMap, vmin=cMin, vmax=cMax)
+    fig.canvas.mpl_connect('key_press_event', process_key)
+    
+def process_key(event):
+    """
+    Processes a key input given by the user
+
+    Parameters
+    ----------
+    event : matplotlib.backend_bases.KeyEvent
+        Key event that is supported by fig.canvas.mpl_connect()
+
+    Returns
+    -------
+    None.
+
+    """
+   
+    fig = event.canvas.figure
+    ax = fig.axes[0]
+    
+    if event.key == 'p':
+        previous_slice(ax)
+        draw(ax,False)
+    elif event.key == 'n':
+        next_slice(ax)
+        draw(ax,False)
+    elif event.key == 'c':
+        change_axis(ax) 
+        draw(ax,True)
+
+    fig.canvas.draw_idle()
+
+def draw(ax,changing):
+    """
+    Draws the new plot once the key input has changed the settings 
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+        Subplot made by matplotlib.
+
+    changing : Boolean
+        A boolean variable that remains False unless we change the axis we are plotting.
+
+    Returns
+    -------
+    None.
+
+    """
+    volume= ax.volume
+    if ax.axis== 0:
+        if changing:
+            del ax.images[0] #this is needed in order to avoid making a new plot in ax.images[1]
+            ax.imshow(volume[ax.index,:,:],cMap, vmin=cMin, vmax=cMax)
+        else:
+            ax.images[0].set_array(volume[ax.index,:,:])
+       
+    elif ax.axis== 1:
+        if changing:
+            del ax.images[0]
+            ax.imshow(volume[:,ax.index,:],cMap, vmin=cMin, vmax=cMax)
+        else:
+            ax.images[0].set_array(volume[:,ax.index,:])
+    
+    elif ax.axis == 2:
+         if changing:
+            del ax.images[0]             
+            ax.imshow(volume[:,:,ax.index],cMap, vmin=cMin, vmax=cMax)
+         else:
+            ax.images[0].set_array(volume[:,:,ax.index])
+       
+
+def previous_slice(ax):
+    """
+    Go to the previous slice.
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+        Subplot made by matplotlib.
+        
+    Returns
+    -------
+    None.
+
+    """
+   
+    volume = ax.volume
+    ax.index = (ax.index - 1) % volume.shape[ax.axis]  # wrap around using %
+    
+def next_slice(ax):
+    """
+    Go to the next slice.
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+        Subplot made by matplotlib.
+
+    Returns
+    -------
+    None.
+
+    """
+    volume = ax.volume
+    ax.index = (ax.index + 1) % volume.shape[ax.axis]
+
+def change_axis(ax):
+    """
+    Changes the axis of view
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+        Subplot made by matplotlib.
+    Returns
+    -------
+    None.
+
+    """
+    volume = ax.volume
+    ax.axis = (ax.axis + 1) % 3
+    ax.index = volume.shape[ax.axis] // 2
+
+# Define the path folder where the file is , or directly the RTdose full file name
```

Comment:
I suggest that the top-level code be indented under a `if __name__ == "__main__":`, so functions could be imported and used from other code.

Reply:
Done